src/components/accounts/CreateAccount.js

```diff
@@ -42,11 +48,11 @@ class CreateAccount extends Component {
             loader: true
         }))
 
-        const { accountId } = this.state
-
-        this.props.createNewAccount(accountId).then(({ error }) => {
-            if (error) return
+        const { accountId, fundingKey } = this.state
 
+        this.props.createNewAccount(accountId, fundingKey).then(({ error }) => {
+            if (error) return;
+                
             this.props.refreshAccount()
```

Comment:
@vgrichina , when creating and funding account using link drop, there’s an issue w/ `refreshAccount` and `history.push(nextUrl)`, throwing the following error: `Error loading account: [-32000] Server error: account testaccount33 does not exist while viewing.` Seems `createNewAccount` doesn’t finish properly before continuing. If I wrap them in a 3s timeout it works fine. Do you have any idea how to solve?

Reply:
@Patrick1904 I think it's bug in nearcore (at minimum there is mismatch of what finality is used for view calls and submitting transactions). Best to create minimum reproducible example in studio. I think it basically needs existing code from wallet, but also add code to view account state immediately after creation:

```
await contract.create_account_and_claim({
             new_account_id: accountId,
             new_public_key: publicKey},
             BOATLOAD_OF_GAS);
const account = await near.account(accountId);
await account.state();
```

Reply:
@vgrichina @Patrick1904 what vlad said is correct. By default view call will execute on final blocks, whereas transaction submission works with the latest head. I am not sure what is needed here exactly, but if you are fine with working with non-final result an easy fix would be just to pass the finality requirement to the rpc (probably not well supported on nearlib side right now. You might need to manually construct the view request). 

Actually what I said above only applies to latest code but not testnet. If you have encountered this on testnet it is probably a separate issue.

Reply:
@bowenwang1996 
>  By default view call will execute on final blocks, whereas transaction submission works with the latest head. 

IMO this should be considered a bug. Default finality should be consistent everywhere.

> probably not well supported on nearlib side right now. 

also would be really nice to not add features to nearcore without adding the same feature on nearlib side

Reply:
@bowenwang1996 also this is for sure happening on testnet right now. I guess we can temporary fix by spinning tx status check in a loop? Will this be too much load on RPC?

Reply:
transactions cannot not work with latest block. That is how it is supposed to work. Is there some repro that I can take a look?

Reply:
> transactions cannot not work with latest block. 

I don't understand what this means.

> Is there some repro that I can take a look?

Yes, this PR is repro. There is a link to running instance on Render (as for every PR). Try creating account with this link:

https://near-wallet-pr-440.onrender.com/create?funding_key 8A2DVKfvobNEyrKLJHcLWbPf5s6T1yf9dEX8kFfzAGgeSTbbRFeUiXNBsH4X23sTRgGo1mtTvQ9CoJZPLDwVE5b

it will redirect you to "sorry all accounts removed" or smth like that – because we couldn't find account after contract method which creates account has finished execution

Reply:
I meant that transactions are naturally applied on top of the latest block of the chain.

Reply:
@bowenwang1996 how does that prevent view calls from operating on previous block? Also maybe it's something else – e.g. send_tx_commit just returns without waiting for all operations to complete when there is create account call from contract?

Reply:
It shouldn't prevent view calls from operating on previous blocks. What can happen is that if you do view on the state that is of the last final block and the block that includes the transaction that creates some account hasn't been finalized yet, this will lead to an error. But I don't think this is what happened given that we don't have this change on testnet yet. I'll take a look soon.

Reply:
Actually the change I made on nearcore side is backward compatible so the problem I described shouldn't exist :)

Reply:
I found the reason after some debugging. The receipts execution results are saved before the head is updated, which means that when querying the state immediately after the `send_tx_commit` returns, it will query the state root as of the previous chunk, not the new chunk which contains the receipts. I don't think this is a bug and you just need to wait maybe 100ms (for the block processing to finish) before querying the state.

Reply:
@bowenwang1996 @vgrichina, seems it needs to wait at least 1000ms before loading account. Barely noticeable from UX perspective but unsure what the threshold actually is and maybe not reliable?

Reply:
>  I don't think this is a bug and you just need to wait maybe 100ms (for the block processing to finish) before querying the state.

@bowenwang1996 this is a bug. We cannot be relying on arbitrary client-side RPC timeouts for system to function properly. We can workaround this to fix current testnet, but it is still a state inconsistency bug on nearcore side.

Reply:
1000ms is pretty big from UX standpoint:
https://www.gigaspaces.com/blog/amazon-found-every-100ms-of-latency-cost-them-1-in-sales/

Reply:
@vgrichina @Patrick1904 actually I was wrong. There is a bug on nearcore side that prevents the request from tracking all receipts result. `send_tx_commit` doesn't actually wait until all receipts are executed before returning. I will try to fix it but before it gets into testnet probably some hackery is required to make things work on wallet side.

Reply:
I submitted a PR to fix this issue https://github.com/nearprotocol/nearcore/pull/2252

Reply:
It's not how it suppose to work. The contract doesn't care about the account being created, that's why the transaction returns the result before the account is being created here:
https://github.com/nearprotocol/near-linkdrop/blob/master/src/lib.rs#L50

The way it should work contract should return the result of the promise creating a new account if it's important for the function. Here is how it should be implemented:
https://github.com/near-examples/pow-faucet/blob/master/contract-rs/src/lib.rs#L76

Notice function `create_account` returns `PromiseOrValue<()>` which means the function may continue execution if it returns a promise. To fix it in linkdrop contract you need to add `.into()` that will convert `Promise` into `PromiseOrValue`
See example here: https://github.com/near-examples/pow-faucet/blob/master/contract-rs/src/lib.rs#L106

Reply:
Here is a linkdrop contract fix: https://github.com/nearprotocol/near-linkdrop/pull/3

Reply:
@bowenwang1996 @vgrichina from comments on #2252 it seems that it's not a nearcore bug? Any thoughts on how we can move forward with this?